src/server/pfs/server/driver.go

```diff
@@ -1060,6 +1060,28 @@ func (d *driver) inspectCommit(ctx context.Context, commit *pfs.Commit, block bo
 	return commitInfo, nil
 }
 
+// resolveCommit is like inspectCommit in that it converts 'commit' (which may
+// be either a commit ID or branch reference) to a repo + commit ID. It accepts
+// an STM so that it can be used in a transaction and avoids an inconsistent
+// call to d.inspectCommit()
+func (d *driver) resolveCommit(stm col.STM, commit *pfs.Commit) error {
+	if len(commit.ID) == uuid.UUIDWithoutDashesLength {
+		return nil // already a commit ID
+	}
+
+	branches := d.branches(commit.Repo.Name).ReadWrite(stm)
+	branchInfo := &pfs.BranchInfo{}
+	// See if we are given a branch
+	if err := branches.Get(commit.ID, branchInfo); err != nil {
+		return nil
+	}
+	if branchInfo.Head == nil {
+		return pfsserver.ErrNoHead{branchInfo.Branch}
+	}
+	commit.ID = branchInfo.Head.ID
+	return nil
+}
```

Comment:
Breaking commit resolution out from `inspectCommit` is an awesome change I've been wanting to make for a while. A few questions:
- it looks like `resolveCommit` doesn't implement the ancestor syntax. I think it probably should, and we should add a test for calls that use `resolveCommit` to use the ancestor syntax if we don't have that yet.
- it looks like `inspectCommit` doesn't use `resolveCommit` but instead resolves the commit itself. Could it just use a call to `resolveCommit` or is it different enough that it needs to do its own resolution.